orchagent/vxlanorch.cpp

```diff
@@ -390,20 +501,36 @@ VxlanTunnelOrch::createNextHopTunnel(string tunnelName, IpAddress& ipAddr, MacAd
         macptr = &mac;
     }
 
-    auto tunnel_obj = getVxlanTunnel(tunnelName);
-
-    sai_object_id_t nh_id, tunnel_id = tunnel_obj->getTunnelId();
-
     if (create_nexthop_tunnel(host_ip, vni, macptr, tunnel_id, &nh_id) != SAI_STATUS_SUCCESS)
     {
         string err_msg = "NH tunnel create failed for " + ipAddr.to_string() + " " + to_string(vni);
         throw std::runtime_error(err_msg);
     }
 
+    //Store the nh tunnel id
+    tunnel_obj->updateNHTunnel(ipAddr, macAddress, vni, nh_id);
+
     SWSS_LOG_INFO("NH vxlan tunnel was created for %s, id 0x%lx", tunnelName.c_str(), nh_id);
     return nh_id;
 }
 
+bool
+VxlanTunnelOrch::removeNextHopTunnel(string tunnelName, IpAddress& ipAddr, MacAddress macAddress, uint32_t vni)
+{
+    SWSS_LOG_ENTER();
+
+    if(!isTunnelExists(tunnelName))
+    {
+        SWSS_LOG_ERROR("Vxlan tunnel '%s' does not exists", tunnelName.c_str());
+        return true;
+    }
+
+    auto tunnel_obj = getVxlanTunnel(tunnelName);
+
+    //Delete request for the nh tunnel id
+    return tunnel_obj->removeNHTunnel(ipAddr, macAddress, vni);
```

Comment:
>removeNHTunnel [](start = 23, length = 14)

since it is already a tunnel object, you do not need tunnel in the function name. -> removeNexthop